data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
                 break;
             } catch (Exception e) {
+                if (!paused) {
+                    paused = true;
+                    consumer.pause(consumer.assignment());
+                }
                 if (e instanceof SizeOverflowException) {
                     topicMetrics.getNumberOfBufferSizeOverflows().increment();
                 } else {
                     LOG.debug("Error while adding record to buffer, retrying ", e);
                 }
                 try {
                     Thread.sleep(100);
+                    consumeRecords();
                 } catch (Exception ex) {} // ignore the exception because it only means the thread slept for shorter time
             }
         }
+        if (paused) {
```

Comment:
When is this executed ? The consumer is still in while loop doing sudo polling ?